lib/fastifySession.js

```diff
@@ -89,7 +89,7 @@ function onSend (options) {
       reply.setCookie(
         options.cookieName,
         session.encryptedSessionId,
-        session.cookie.options()
+        session.cookie.options(getRequestProto(request))
```

Comment:
I think we should also do some other checks besides just the requestProto header, if possible:
See the issecure function in express-session:
https://github.com/expressjs/session/blob/b22384b712fea118f1c3eb5b0d79312ebd25e97c/index.js#L612

We may also want a way to check a custom header (e.g. X-Forwarded-Scheme instead of X-Forwarded-Proto, in the case that X-Forwarded-Proto header is overwritten by some ELBs such as AWS ALB)

Can we add another option?
`cookie.secureHeaderName`

e.g. If cookie.secureAutoHeader = 'X-Forwarded-Scheme', instead of checking `X-Forwarded-Proto` header, we will check the header `X-Forwarded-Scheme' instead to determine the value for 'secure' when set to 'auto'

Reply:
Adding secureAutoHeader is too custom here. A solution for that would be to add a function for the auto behavior but it's out of scope here IMO

Reply:
Okay that's fine, thanks